samwise.js

```diff
@@ -3,69 +3,63 @@
  *
```

Comment:
SCRIPT-Nice job on the comprehensive test coverage, it gives a lot of confidence in the stability of the new feature.